RecoVertex/BeamSpotProducer/plugins/BeamSpotOnlineProducer.cc

```diff
@@ -79,7 +80,7 @@ void BeamSpotOnlineProducer::fillDescriptions(edm::ConfigurationDescriptions& iD
   ps.add<double>("maxZ", 40.);
   ps.add<double>("setSigmaZ", -1.);
   ps.addUntracked<unsigned int>("beamMode", 11);
-  ps.add<InputTag>("src", InputTag("hltScalersRawToDigi"));
+  ps.addOptional<InputTag>("src", InputTag("hltScalersRawToDigi"))->setComment("SCAL decommissioned after Run 2");
```

Comment:
Just to see if I understand (maybe I don't):

 - [1] works in the same way with `add` and `addOptional` (with `add`, `src` is set to the default `"hltScalersRawToDigi"`, which is anyway ignored);
 - [2] works with `add` if the `"hltScalersRawToDigi"` product is available/valid (`src` is 'silently' set to its default `"hltScalersRawToDigi"`), while it fails at runtime with `addOptional` (as `src` is missing in the PSet).

The net result is that `addOptional` requires the user to explicitly specify `src` in the PSet when `useTransientRecord = False`. Is that right? Is that the intention?

[1]
```py
hltOnlineBeamSpot = cms.EDProducer( "BeamSpotOnlineProducer",
#    src = cms.InputTag( "hltScalersRawToDigi" ),
    useTransientRecord = cms.bool( True ),

    changeToCMSCoordinates = cms.bool( False ),
    maxZ = cms.double( 40.0 ),
    setSigmaZ = cms.double( 0.0 ),
    beamMode = cms.untracked.uint32( 11 ),
    gtEvmLabel = cms.InputTag( "" ),
    maxRadius = cms.double( 2.0 )
)
```

[2]
```py
hltOnlineBeamSpot = cms.EDProducer( "BeamSpotOnlineProducer",
#    src = cms.InputTag( "hltScalersRawToDigi" ),
    useTransientRecord = cms.bool( False ),

    changeToCMSCoordinates = cms.bool( False ),
    maxZ = cms.double( 40.0 ),
    setSigmaZ = cms.double( 0.0 ),
    beamMode = cms.untracked.uint32( 11 ),
    gtEvmLabel = cms.InputTag( "" ),
    maxRadius = cms.double( 2.0 )
)
```

Reply:
> The net result is that addOptional requires the user to explicitly specify src in the PSet when useTransientRecord = False. Is that right? Is that the intention?

Yes, indeed.